transport/x/grpc/headers.go

```diff
@@ -162,6 +162,9 @@ func addApplicationHeaders(md metadata.MD, headers transport.Headers) error {
 
 // getApplicationHeaders returns the headers from md without any reserved headers.
 func getApplicationHeaders(md metadata.MD) (transport.Headers, error) {
+	if md == nil {
```

Comment:
Any reason not to do `len(md) == 0` instead?

Reply:
Doing that instead.